test/e2e/run.sh

```diff
@@ -802,6 +802,29 @@ $py_assertion
     return 0
 }
 
+kubectl-force-delete() { # script API
+    # Usage: kubectl-force-delete OBJECT NAME
+    #
+    # Examples:
+    # - kubectl-force-delete namespace my-namespace
+    #     Deletes a namespace that is stuck in the "Terminating" phase.
+    if [[ "$1" == "namespace" ]] || [[ "$1" == "ns" ]]; then
+        local ns="$2"
+        vm-command "
+            kubectl get namespace $ns -o json > force-delete-ns.json || exit 0
+            (
+              grep -E phase.*Terminating force-delete-ns.json || exit 0
+              sed -i '/\"kubernetes\"/d' force-delete-ns.json
+              pgrep -f 'kubectl proxy --port 8001' || ( kubectl proxy --port 8001 </dev/null >/dev/null 2>&1 & sleep 3 )
+              curl -sk -H 'Content-Type: application/json' -X PUT --data-binary @force-delete-ns.json http://127.0.0.1:8001/api/v1/namespaces/$ns/finalize
```

Comment:
nitpick: Is it really so obvious (even after half a year down the road) what is going on here, that it does not warrant a comment explaining it ?

Reply:
Otherwise LGTM.

Reply:
I was reading about the same topic some time ago and the answer is some resources are left in that namespace and it blocks the namespace controller to delete it.  The clean way was to find those resources for deletion: `kubectl api-resources --verbs=list --namespaced -o name   | xargs -n 1 kubectl get --show-kind --ignore-not-found -n` and then the namespace controller kicks in correctly. Source: https://stackoverflow.com/questions/52369247/namespace-stuck-as-terminating-how-i-removed-it

Reply:
@mythi:
```
$ kubectl create namespace delme
namespace/delme created
$ kubectl api-resources --verbs=list --namespaced -o name | while read obj; do echo --- $obj:; kubectl get --show -kind --ignore-not-found -n delme $obj; done
error: unable to retrieve the complete list of server APIs: metrics.k8s.io/v1beta1: the server is currently unable to handle the request
--- configmaps:                                                                                                                                                      NAME                         DATA   AGE
configmap/kube-root-ca.crt   1      9s
--- endpoints:
--- events:
--- limitranges:
--- persistentvolumeclaims:
--- pods:
--- podtemplates:
--- replicationcontrollers:
--- resourcequotas:
--- secrets:
--- serviceaccounts:                                                                                                                                                 NAME                     SECRETS   AGE
serviceaccount/default   0         11s
--- services:
--- controllerrevisions.apps:
--- daemonsets.apps:
--- deployments.apps:
--- replicasets.apps:
--- statefulsets.apps:
--- horizontalpodautoscalers.autoscaling:
--- cronjobs.batch:
--- jobs.batch:
--- ciliumendpoints.cilium.io:
--- ciliumnetworkpolicies.cilium.io:
--- leases.coordination.k8s.io:
--- adjustments.criresmgr.intel.com:
--- endpointslices.discovery.k8s.io:
--- events.events.k8s.io:
--- ingresses.networking.k8s.io:
--- networkpolicies.networking.k8s.io:
--- poddisruptionbudgets.policy:
--- rolebindings.rbac.authorization.k8s.io:
--- roles.rbac.authorization.k8s.io:
--- csistoragecapacities.storage.k8s.io:
$ kubectl delete namespace delme
namespace "delme" deleted
```
...and after this it's stuck. 
```
$kubectl get namespaces
NAME                  STATUS        AGE
btype1ns0             Active        16h
default               Active        36d
delme                 Terminating   4m52s
```

I have exactly the same list of resources in another cluster (both running Kubernetes 1.24.1), and there the namespace controller handles the deletion without getting stuck. I haven't installed and uninstalled the metrics  in the working cluster, but I have in the broken cluster. Maybe my uninstall was unclean, because there's the error - which may or may not be related to namespace controller problems, too.

Reply:
It seems this goes back to @klihub question that do we know what's going on :-)  FWIW, I have no objections to the change itself. Just wanted to add a comment what I learnt myself recently.